File: DarkOrbit 10.0/Net/netty/commands/PetGearSelectCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class PetGearSelectCommand
    {
        public const short ID = 18610;

        public static byte[] write(PetGearTypeModule gearType, List<int> optionalParams)
        {
            var param1 = new ByteArray(ID);
            param1.write(gearType.write());
            param1.writeShort(330);
            param1.writeInt(optionalParams.Count);
            foreach(var i in optionalParams)
            {
                param1.writeInt(i << 3 | i >> 29);
            }
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/PetGearSelectCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;
import java.util.Vector;

import simulator.netty.ServerCommand;

/**
Created by LEJYONER on 09/01/2018.
*/

public class PetGearSelectCommand
        implements ServerCommand {

    public static int ID = 31289;

    public PetGearTypeModule gearType;   
    public Vector<Integer> optionalParams;
   
    public PetGearSelectCommand(PetGearTypeModule param1, Vector<Integer> param2) {
    	this.gearType = param1;
    	this.optionalParams = param2;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeInt(this.optionalParams.size());
            for (Integer i : this.optionalParams) {
                param1.writeInt(i >>> 16 | i << 16);
            }
            this.gearType.write(param1);
            param1.writeShort(6186);
            param1.writeShort(710);
        } catch (IOException e) {
        }
    }

}
